eos_data_distribution/avahi_monitor.py

```diff
@@ -188,6 +192,12 @@ def remove_nexthop(self, faceUri):
         return check_call(["nfdc", "remove-nexthop",
                            str(SUBSCRIPTIONS_SOMA), faceUri])
 
+    def start_edge(self):
+        return check_call(["systemctl", "start", "edd-soma-subscriptions-producer"])
```

Comment:
There’s nothing stopping edd-ssp doing more than one thing well. Tying the policy of whether to advertise SOMA subscriptions with systemd’s run status for edd-ssp is conflating two things, which will result in confusion when one of them goes wrong. What if edd-ssp crashes at the same time as the policy decision from the Avahi service changes — systemd will unconditionally restart edd-ssp, but that might leave it running when the Avahi service doesn’t want it there. What if we add another component to endless-ndn which also has a say in the policy for running edd-ssp? Now both that component and the Avahi service need to communicate to work out whether to call `systemctl start` or `systemctl stop`. What if the initial call to `systemctl start` from the Avahi service fails? Should it start a timer and periodically retry starting edd-ssp? If so, that’s partially reinventing systemd’s job.